google-cloud-bigquery/test/google/cloud/bigquery/dataset_reference_update_test.rb

```diff
@@ -123,4 +123,23 @@
     _(dataset.default_encryption).must_be :frozen?
     mock.verify
   end
+
+  it "updates its storage_billing_model" do
+    storage_billing_model = "LOGICAL"
+
+    mock = Minitest::Mock.new
+    bigquery.service.mocked_service = mock
+    mock.expect :get_dataset, dataset_gapi, [project, dataset_id]
+    updated_gapi = dataset_gapi.dup
+    updated_gapi.storage_billing_model = storage_billing_model
+    patch_gapi = Google::Apis::BigqueryV2::Dataset.new storage_billing_model: storage_billing_model, etag: dataset_gapi.etag
```

Comment:
since we are doing a dataset_gapi.dup for updated_gapi, it should already contain the etag (which you are setting to the same value). why do we need to set it explicitly

Reply:
Yes, `updated_gapi` contains etag, but `patch_gapi` is reconstructed from constructor. And `patch_gapi` is what is sent to the backend. So we need to pass it here during initialization.

I tried to remove it, the but the test failed (due to the above reason).